packages/astro/src/compiler/codegen/index.ts

```diff
@@ -716,6 +731,11 @@ async function compileHtml(enterNode: TemplateNode, state: CodegenState, compile
                   importStatements.add(wrapperImport);
                 }
               }
+              if (hydrationAttributes.method === 'only') {
+                // Remove component imports for client-only components
+                const componentImports = state.componentImports.get(componentName) || [];
+                componentImports.map((componentImport) => state.importStatements.delete(componentImport));
```

Comment:
Clean way to do this!

Reply:
Thanks!  I _may_ have completely broken markdown rendering before realizing it'd be simpler to just remove imports after the fact :laughing: